src/services/AmazonS3Service.ts

```diff
@@ -0,0 +1,75 @@
+import axios, { AxiosError, AxiosResponse } from 'axios';
+
+interface Image {
+  imageBinary: string;
+  imageType: string;
+}
+
+class AmazonS3Service {
+  onFileChange(e: React.ChangeEvent<HTMLInputElement> | DragEvent): Promise<Image> {
+    // Get the file from selection or drag and drop
+    const files =
+      (e as React.ChangeEvent<HTMLInputElement>)?.target?.files ||
+      (e as DragEvent)?.dataTransfer?.files ||
+      [];
+    if (!files.length) return Promise.resolve({ imageBinary: '', imageType: '' });
+    const file = files[0];
+    // Convert file into binary string
+    const reader = new FileReader();
+    return new Promise((resolve, reject) => {
+      reader.onload = (e: ProgressEvent<FileReader>) => {
+        // Only allow jpg, jpeg, png, and gifs to be selected
+        const file = e?.target?.result as string;
+        const fileRegexMatches = file.match(/image\/(jpe?g|png|gif)/);
+        if (fileRegexMatches === null) {
+          return reject(new Error('Wrong file type. Only images are allowed.'));
+        }
+        // Restrict the size of the file
+        if (file.length > parseInt(process.env.REACT_APP_AWS_MAX_IMAGE_SIZE as string)) {
```

Comment:
I was just thinking that because this is all client facing, we should try to restrict users from being allowed to change how large  the image size can be. But then again, the user could potentially just remove this line too, so it might not make a big difference.

Reply:
Ah sorry, just taking a look at this now! That makes sense, yeah we can keep it then :)